# Week 2 – SQL for Data Analysis

In [13]:
import pandas as pd
import sqlite3

In [14]:
df = pd.read_excel("SQL_Sales_Dataset_200_Rows.xlsx")
df.head()

,order_id,customer_name,order_date,category,sub_category,product_name,quantity,unit_price,total_price,region
0,1000,Caleb Davis,2025-01-31,Clothing,Jeans,Jeans Try,2,1197,2394,West
1,1001,Misty Herrera,2025-05-05,Clothing,Jeans,Jeans New,9,1509,13581,South
2,1002,Rick Soto,2025-06-09,Clothing,Shoes,Shoes Us,5,1046,5230,East
3,1003,Sean Haas,2025-02-27,Furniture,Desk,Desk Section,10,292,2920,South
4,1004,Ann Hall,2025-06-01,Furniture,Cabinet,Cabinet Center,2,548,1096,North


In [15]:
conn = sqlite3.connect("sales_database.db")

df.to_sql("sales", conn, if_exists="replace", index=False)

print("SQL database created successfully!")

SQL database created successfully!


In [16]:
query = "SELECT * FROM sales LIMIT 5;"

result = pd.read_sql_query(query, conn)
result

,order_id,customer_name,order_date,category,sub_category,product_name,quantity,unit_price,total_price,region
0,1000,Caleb Davis,2025-01-31 00:00:00,Clothing,Jeans,Jeans Try,2,1197,2394,West
1,1001,Misty Herrera,2025-05-05 00:00:00,Clothing,Jeans,Jeans New,9,1509,13581,South
2,1002,Rick Soto,2025-06-09 00:00:00,Clothing,Shoes,Shoes Us,5,1046,5230,East
3,1003,Sean Haas,2025-02-27 00:00:00,Furniture,Desk,Desk Section,10,292,2920,South
4,1004,Ann Hall,2025-06-01 00:00:00,Furniture,Cabinet,Cabinet Center,2,548,1096,North


In [17]:
query = """
SELECT customer_name, SUM(total_price) AS total_spent
FROM sales
GROUP BY customer_name
ORDER BY total_spent DESC
LIMIT 10;
"""

result = pd.read_sql_query(query, conn)
result

,customer_name,total_spent
0,Lynn Garrison,47940
1,Debbie Turner,44990
2,Megan Charles,44620
3,Michelle Beltran,42471
4,Rick Sanford,41211
5,Patrick Watson,37680
6,Christopher Meyer,37320
7,Todd Barnes,36152
8,Stephanie Martinez,34576
9,Kimberly Alvarado,31752


In [18]:
query = """
SELECT AVG(total_price) AS average_order_value
FROM sales;
"""

result = pd.read_sql_query(query, conn)
result

,average_order_value
0,12100.535


## Assignment Results

### Top 10 Customers
The query identifies the top 10 customers based on their total spending.

### Average Order Value
The average order value is **12,100.535**.